### PR TITLE
Refactor/crs label select logic

### PR DESCRIPTION
**Pull request by tomersf**
Removed the need for an email label in the user role cr, by adding filter logic of the username.

Fixed a small type check of go in the infinite for loop with the select statement.

### CODE REVIEW

File: internal/k8s/sync.go [hunk 1]
```diff
@@ -230,8 +232,7 @@ func (client *Client) GetUser(email string) (UserResource, error) {
 
 func (client *Client) AssertPassword(username string, password string) bool {
 	logger := misc.GetLogger()
-	username = strings.Replace(username, "@", "__at__", 1)
-	userCR, _ := getCRForGroupKindWithLabel("iam-backbone.org", "backboneuser", "email", username, client.k8sDynClient)
+	userCR, _ := getCRForGroupKindWithUsername("iam-backbone.org", "backboneuser", username, client.k8sDynClient)
```
Comment (omrico) [inline internal/k8s/sync.go:235]: It's probably better to decouple the k8s abstraction from the business logic here (and everywhere else, but that's for another time), in other words, delete this getCRForGroupKindWithLabel() method and instead in the  AssertPassword(), create a private method that filters CRs by username.
Call getCRsForGroupKind("iam-backbone.org", "backboneuser"...) , apply the filter, convert and return the User model.

Comment (tomersf) [reply internal/k8s/sync.go:235]: I'm not sure that I fully understand.
Do you want to remove the cr fetch logic from the AssertPassword completely? and pass the CR from outside to it as a parameter?

and when u write create private method as in on the Client struct or just as it is now (getCRForGroupKindWithUsername).

I can change the logic inside of it to call getCRsForGroupKind and filter based on username

Comment (tomersf) [reply internal/k8s/sync.go]: Added the method and did a small refactor in the labels method for checking the length of crs list

File: internal/k8s/sync.go [hunk 2]
```diff
@@ -228,17 +231,30 @@ func (client *Client) GetUser(email string) (UserResource, error) {
...
 func (client *Client) AssertPassword(username string, password string) bool {
 	logger := misc.GetLogger()
-	username = strings.Replace(username, "@", "__at__", 1)
-	userCR, _ := getCRForGroupKindWithLabel("iam-backbone.org", "backboneuser", "email", username, client.k8sDynClient)
-	specData, _ := json.Marshal(userCR.Object["spec"])
-	var userWithPassword UserPasswordResource
-	_ = json.Unmarshal(specData, &userWithPassword)
+	userWithPassword, err := getUserWithPasswordByUsername(username, client.k8sDynClient)
+	if err != nil {
+		logger.Warnf("%s", err.Error())
```
Comment (omrico) [inline internal/k8s/sync.go:252]: Need clearer message

File: internal/k8s/sync.go [hunk 3]
```diff
@@ -209,14 +209,17 @@ func getCRForGroupKindWithLabel(group string, kind string, labelName string, lab
 	// Label selector to filter CRs
 	labelSelector := labels.SelectorFromSet(labels.Set{labelName: labelValue})
 
-	// Fetch the list of CRs with the specified label
+	// // Fetch the list of CRs and filter by username
```
Comment (omrico) [inline internal/k8s/sync.go:212]: nit: redundant //

File: internal/k8s/sync.go [hunk 4]
```diff
@@ -198,7 +198,7 @@ func getCRForGroupKind(group string, kind string, crName string, dynClient *dyna
 	return cr, nil
 }
 
-func getCRForGroupKindWithLabel(group string, kind string, labelName string, labelValue string, dynClient *dynamic.DynamicClient) (*unstructured.Unstructured, error) {
+func getCRForGroupKindWithLabel(group, kind, labelName, labelValue string, dynClient *dynamic.DynamicClient) (*unstructured.Unstructured, error) {
```
Comment (omrico) [inline internal/k8s/sync.go:201]: nice!

File: internal/k8s/sync.go (hunk 2 above)
Comment (tomersf) [reply internal/k8s/sync.go]: Done

File: internal/k8s/sync.go (hunk 3 above)
Comment (tomersf) [reply internal/k8s/sync.go]: Done

File: internal/k8s/sync.go (hunk 4 above)
Comment (tomersf) [reply internal/k8s/sync.go:201]: Thanks 😊

**Review by omrico — APPROVED** [review]

**Pull request merged**